### ISSUE TITLE
personalize sidebar

### ISSUE THREAD

**Issue by werererer** [TeamNewPipe/NewPipe#4081]
#### Describe the feature you want

It would be great if there would be a feature to personalize the left sidepanel (like removing the hot section or changing the order of the section etc.)

#### How will you/everyone benefit from this feature?

As I am too addicted to pointless youtube videos clicking the hot section is very tempting. With the help of this feature I wouldn't need that much willpower to withstand wasting my time.

**Comment by Stypox** [comment]
See #3343 and #2319

**Comment by opusforlife2** [comment]
I think you should close 3343 in favour of _this_ one, instead. Both 2319 and 3343 are _subsets_ of this one.

**Comment by Stypox** [comment]
Bounty of $30 by @sulyniv in #2319

**Comment by Gymcap** [comment]
To add to this, theres a handful of services I'd like to see in NewPipe that I don't think would fit as defaults on the side panel for most, i.e. Vimeo and DailyMotion.

I feel this kind of feature would also open the way for the possibily of more niche services without bothering users who don't use them.

**Comment by opusforlife2** [comment]
@Gymcap #4054 would let the user _choose_ which extension they want to install, putting that power completely in their hands.

**Comment by wb9688** [comment]
However that's only for adding other streaming services, there currently are no plans for other kinds of extensions.

**Comment by werererer** [comment]
I experimented with the code the past hours and may even be able to implement it. If it works i will make a PR in the next few hours/days

**Comment by werererer** [comment]
I have successfully written this feature. Need to clean up my code and make a PR today/tomorrow if everything works out

**Comment by werererer** [comment]
PR request: https://github.com/TeamNewPipe/NewPipe/pull/4104

**Comment by opusforlife2** [comment]
Pull request request? ;)

@Stypox Isn't the 'PR open' label redundant? Linked pull requests show up in the Issue list directly.

**Comment by werererer** [comment]
I Dont get it:/

**Comment by opusforlife2** [comment]
PR is Pull Request. PR request is Pull Request request. Lol, ignore me, I'm just nitpicking.

**Comment by werererer** [comment]
Ah ok xD

**Comment by Stypox** [comment]
@opusforlife2 yeah, the "PR open" label was useful before Github introuced linked issues, now it is mostly useless and could be removed

**Comment by SoerenDaenekas** [comment]
for me the point of newpipe is to be non distracting, no ads and to have high usuability.

Defaulting to the trends page, where videos are shown that I hate, makes me not wanting to use this app. While all features the app provides are excellent, i am not willing to use it, because opening it forces me to see the thumbnail of videos that annoy me a lot.

**Comment by bademux** [comment]
@Stypox  how about rebrand the feature: removing trending and suggested videos is ... **whitelisting channels** feature that a lot of people ask for.
Just by itself it has great value as parent control.
Add password protection to Settings menu and you will have solid application for Kids and Teens - something that google ignores for years.

**Comment by Guigui220D** [comment]
I understand that newpipe wants a more general way to remove the trendings tab than just an option to remove the trending tab but this makes the problem much more complicated than it is and i know a lot of people would like a way to hide the trending tab... Is there any chance this gets implemented some day? The fact that all issues about hiding trending are closed in favor of this much more complicated and vague issue makes it stall... It would be nice to at least have a workaround until this more genral feature is here

**Comment by Guigui220D** [comment]
> @Guigui220D You can already remove any tab from the homepage. There is complete freedom about which tabs you want and in what order. Have you not gone through the settings?
> 
> This issue is only about modifying the sidebar.

I meant the treding button in the sidebar...

**Comment by Guigui220D** [comment]
I feel like it could be discussed outside of a PR, since all users arent necessarily tech-savvy enough to implement a fix
I'd open an issue about it but then it'd be closed in favor of this probably

**Comment by pc386** [comment]
I can't believe this is still not done after 4 years? What is the holdup?

**Comment by throwaway242685** [comment]
one of the latest updates implemented way too many tabs:

<img alt="screenshot_20250820-133521.png" src="https://github.com/user-attachments/assets/73c395f6-aa76-4d53-9c70-e0509508f2a8" width=284 />


that's 14 tabs:(

it becomes really uncomfortable to try to reach "Settings" when you're using your phone in the horizontal orientation.

I keep misclicking the wrong tabs all the time.

can this feature please be considered?

I literally only use 3 out of 14 tabs... (playlists, history, settings)

**Comment by krwow** [comment]
> one of the latest updates implemented way too many tabs:
> <img alt="screenshot_20250820-133521.png" width="284" src="https://private-user-images.githubusercontent.com/85243826/480125195-73c395f6-aa76-4d53-9c70-e0509508f2a8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Nzc3MjY4ODYsIm5iZiI6MTc3NzcyNjU4NiwicGF0aCI6Ii84NTI0MzgyNi80ODAxMjUxOTUtNzNjMzk1ZjYtYWE3Ni00ZDUzLTljNzAtZTA1MDk1MDhmMmE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNTAyVDEyNTYyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhYzAzZmFiN2FhM2M1YjFhMTk0MzIyNTM1MzQyNjRkNDJmZTMxM2YxZWMwY2RlZTY4YWExMjY3ZGZmMjdkNDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.oYBVTADkfPfHHNn3LimoKJIDklgeVC_e74os895E9xo">
> 
> (look how the scroll bar shows up even in the vertical orientation 😐)
> 
> that's 14 tabs...
> 
> it becomes really uncomfortable to try to reach "Settings" when you're using your phone in the horizontal orientation.
> 
> I keep misclicking the wrong tabs all the time.
> 
> can this feature please be considered?
> 
> I literally only use 3 out of 14 tabs... (playlists, history, settings)

Unfortunately nothing has changed since the last comment.

Adding an option to hide some of unneccessary items in the sidebar would make a whole experience less distracting.
